framework/pages/documentViewer/comment.ts

```diff
@@ -0,0 +1,60 @@
+import {clickOnElement, clickOnElementCustomSelector,  waitForVisible, fillInInput, waitForText, fillInputCustomSelector} from "../../helpers/utils.ts";
+import {expect, type Page} from "@playwright/test";
+
+const commentText = 'Amazing Document To Comment On!!!!';
+const  commentTextArea = 'comment-textarea';
+const  removeCommentButton = 'remove-comment-button';
+const  submitCommentButton = 'submit-comment-button';
+const  firstComment = 'sidebar-comments-comment-0';
+const  replyText = 'This is reply to the first comment!!!!';
+const  commentReplyOpen = 'comment-reply-open';
+const  commentReplyClose = 'comment-reply-close';
+const  confirmRemoveCommentButton = 'confirm-remove-comment-button';
+const commentDeleteToast = 'Comment successfully deleted';
+const replyOnFirstComment = '//*[@data-test-selector="sidebar-comments-comment-0"]//*[@data-test-selector = "comment-textarea"]'
+const replyComment = 'sidebar-comments-comment-0-replies-comment-0'
+const replySubmitButton = '//*[@data-test-selector="sidebar-comments-comment-0"]//button[@data-test-selector="submit-comment-button"]'
+const toastMessage = '//*[@data-test-selector="toast-message"]';
+const noCommentCTAOnLoggedOutUser = '//*[@data-test-selector="login-register-to-comment-box"]';
+
+const typeAndSubmitComment = async (page: Page) => {
+    await waitForVisible(page, commentTextArea);
+    await fillInInput(page, commentTextArea, commentText);
+    await clickOnElement(page, submitCommentButton);
+}
+
+const cancelOnComment = async (page: Page) => {
+     await waitForText(page, commentText);
+     await clickOnElement(page, firstComment);
+     await clickOnElement(page, commentReplyOpen);
+     await clickOnElement(page, commentReplyClose);
+}
+
+const replyOnComment = async (page: Page) => {
+    await waitForVisible(page, firstComment );
+    await waitForText(page, commentText);
+    await clickOnElement(page, commentReplyOpen);
+    await fillInputCustomSelector(page, replyOnFirstComment, replyText);
+    await clickOnElementCustomSelector(page, replySubmitButton);
+}
+
+const deleteComment = async (page: Page) => {
+    await clickOnElement(page, firstComment);
+    await clickOnElement(page, removeCommentButton);
+    await clickOnElement(page, confirmRemoveCommentButton);
+    await expect(page.locator(toastMessage)).toHaveText(commentDeleteToast);
+}
+
+const deleteReply = async (page: Page)=> {
+    await clickOnElement(page, replyComment);
+    await clickOnElement(page, removeCommentButton);
+    await clickOnElement(page, confirmRemoveCommentButton);
```

Comment:
typo here `confirmRemoveCommentButton`. This selector is for deleting comment, not reply.

```suggestion
   await clickOnElementCustomSelector(page, removeReplyOnCommentButton);
```